### PR TITLE
In-2616 allow filtering based on a linked deal on events list

### PR DESCRIPTION
**Pull request by EOSullivanBerlin**
This is need for the new deal detail page. 
We want to show our customers the events that are linked to that deal.

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -372,6 +372,10 @@ We consider the following changes to be backwards-incompatible:
 We list all backwards-compatible additions here. These are currently available in all published versions.
 (There is also a [list of backwards-incompatible upgrades](#changelog) available, but those only apply if you [upgrade your API version](#upgrading-your-api-version).)
 
+#### June 2020
```
Comment (luisaterrabrito) [inline apiary.apib]: I think we call it Latest usually

**Review by lorgan3 — DISMISSED** [review]

File: apiary.apib (hunk 1 above)
Comment (EOSullivanBerlin) [reply apiary.apib]: Okay I will change it to latest.

**Review by lorgan3 — DISMISSED** [review]

**Review by lowiebenoot — DISMISSED** [review]

File: src/04-calendar/events.apib [hunk 2]
```diff
@@ -28,6 +28,7 @@ Get a list of calendar events.
                     + Members
                         + contact
                         + company
+                        + deal
```
Comment (dietrichm) [inline src/04-calendar/events.apib:31]: This exposes that we will start having `deal` type links, so we can add that entity type to the enum of `links` types on lines 78 and 122.

File: src/changes-backwards-compatible.apib [hunk 3]
```diff
@@ -4,6 +4,10 @@
 We list all backwards-compatible additions here. These are currently available in all published versions.
 (There is also a [list of backwards-incompatible upgrades](#changelog) available, but those only apply if you [upgrade your API version](#upgrading-your-api-version).)
 
+#### Latest
+
+- We added filtering on a linked `deal` to `events.list`.
```
Comment (dietrichm) [inline src/changes-backwards-compatible.apib]: PP: let's make it explicit that the events endpoints now have support for deal links, so we added `link` as a link type plus allow to filter on it.

**Comment by dietrichm** [comment]
There's some overlap with https://github.com/teamleadercrm/api/pull/507. I propose to merge that one and rebase this PR.

**Comment by EOSullivanBerlin** [comment]
@dietrichm You are right the branches are very similar.

**Review by lowiebenoot — APPROVED** [review]

**Review by lorgan3 — APPROVED** [review]

**Review by lowiebenoot — APPROVED** [review]

File: src/04-calendar/events.apib (hunk 2 above)
Comment (dietrichm) [reply src/04-calendar/events.apib:31]: What is the reason that we only have the new `deal` link type on the list endpoint and not on the info one?

File: src/changes-backwards-compatible.apib [hunk 4]
```diff
@@ -4,6 +4,10 @@
 We list all backwards-compatible additions here. These are currently available in all published versions.
 (There is also a [list of backwards-incompatible upgrades](#changelog) available, but those only apply if you [upgrade your API version](#upgrading-your-api-version).)
 
+#### Latest
+
+- We added filtering on a new `link` type `deal` to `events.list`.
```
Comment (dietrichm) [inline src/changes-backwards-compatible.apib:9]: Filtering is not the main takeaway I think. Might be better to word this as:
```suggestion
- As events can be linked to deals, we added the `deal` link type to `events.list` and `events.info`, and allow filtering on deals in `events.list`.
```

**Review by dietrichm — COMMENTED** [review]
As #507 got merged, can we rebase this PR to master and fix the missing deal link type for the info endpoint?

File: src/04-calendar/events.apib (hunk 2 above)
Comment (EOSullivanBerlin) [reply src/04-calendar/events.apib:31]: We are only using it on the list. Does it make sense to add it to an end point if we are not using it. I assume this is a case of YAGNI You aren't going to need it.

File: src/changes-backwards-compatible.apib (hunk 4 above)
Comment (EOSullivanBerlin) [reply src/changes-backwards-compatible.apib:9]: I will use your description but not mention the info as we have not added that yet.

File: src/changes-backwards-compatible.apib (hunk 3 above)
Comment (EOSullivanBerlin) [reply src/changes-backwards-compatible.apib]: Okay

**Pull request closed without merging**